pyproject.toml

```
@@ -28,6 +28,12 @@ matplotlib = "^3.9.2"
 pylint = "^3.3.1"
 notebook = "^7.2.2"
 
+[tool.poetry.requires-plugins]
+poetry-dynamic-versioning = { version = ">=1.0.0,<2.0.0", extras = ["plugin"] }
+
+[tool.poetry-dynamic-versioning]
+enable = true
```

Consider specifying further what the style and version control types are here. 

```suggestion
enable = true
style = "pep440"
vcs = "git"
```

---

Added!